### ISSUE TITLE
monitor progress of lbfgs_minimize

### ISSUE THREAD

**Issue by APJansen** [tensorflow/probability#1069]
I would like to monitor the intermediate states of the L-BFGS optimization. One way to do this is to attach an attribute to the loss function that is passed to it, e.g. `loss_function.loss = []` and inside the definition of `loss_function` append to that every time it is called. 
However, since it is called multiple times during the line search stage, this will record all of those, many of which result in an increasing loss. 

Is there any way to only monitor every L-BFGS step, excluding the calls used for line search? Perhaps some internal flag that can be used, or an entirely different way?
